Add documentation (#56)
MatterMiners/tardis: docs/source/plugins/plugins.rst

=======
Plugins
=======

> All plugins are configured within the `plugins` section of the TARDIS configuration. Using multiple plugins are

Reviewer: ```suggestion
All plugins are configured within the `plugins` section of the TARDIS configuration. Using multiple plugins is```

supported by using a separate MappingNode per plugin.

.. code-block:: yaml

    Plugins:
        Plugin_1:
            option_1: my_option_1
        Plugin_2:
            option_123: my_option_123

> Sqlite Registry

Reviewer: ```suggestion
SQLite Registry
```

---------------
The :py:class:`~tardis.plugins.sqliteregistry.SqliteRegistry` implements a persistent storage of all Drone states in a

> Sqlite database. The usage of this module is recommended in order to recover the last state of TARDIS in case the

Reviewer: ```suggestion
SQLite database. The usage of this module is recommended in order to recover the last state of TARDIS in case the
```

service has to be restarted.

Available configuration options
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

+----------------+-----------------------------------+-----------------+

> | Option         | Short Description                 | Optionality     |

Reviewer: ```suggestion
| Option         | Short Description                 | Requirement     |
```

+================+===================================+=================+

> | db_file        | Location of the Sqlite database.  |  **Required**   |

Reviewer: ```suggestion
| db_file        | Location of the SQLite database.  |  **Required**   |
```

+----------------+-----------------------------------+-----------------+

Example configuration
~~~~~~~~~~~~~~~~~~~~~

.. code-block:: yaml

    Plugins:
      SqliteRegistry:
        db_file: drone_registry.db

Telegraf Monitoring
-------------------
The :py:class:`~tardis.plugins.telegrafmonitoring.TelegrafMonitoring` implements an interface to monitor state changes

> of the Drones in a telegraf service running an UDP input module.

Reviewer: ```suggestion
of the Drones in a telegraf service running a UDP input module.
```

Available configuration options
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

+----------------+---------------------------------------------------------------------------+-----------------+

> | Option         | Short Description                                                         | Optionality     |

Reviewer: ```suggestion
| Option         | Short Description                                                         | Requirement     |
```

+================+===========================================================================+=================+

> | host           | Hostname or IP address of the telegraf UDP input module is listening to.  |  **Required**   |

Reviewer: ```suggestion
| host           | Hostname or IP address the telegraf UDP input module is listening to.     |  **Required**   |
```

+----------------+---------------------------------------------------------------------------+-----------------+
| port           | Port the telegraf UDP input module is listening on.                       |  **Required**   |
+----------------+---------------------------------------------------------------------------+-----------------+

> | default_tags   | Tags that should included by default for all entries send to telegraf.    |  **Optional**   |

Reviewer: ```suggestion
| default_tags   | Tags that should be included by default for all entries sent to telegraf. |  **Optional**   |
```
